dynamic_programming/dp_bitmasking.cpp

```diff
@@ -0,0 +1,132 @@
+/**
+ * @file
+ * @brief Functions for [Bitmasking DP
+ * approach](https://codeforces.com/blog/entry/337) implementation
+ *
+ * @details
+ * Given Integer N and Array of size 2$\cdot${N}. Make N pairs using each
+ * element exactly once such that following score is maximized.
+ * \[ score = \sum_{i=1}^{n} gcd(x_{i}, y_{i}) \]
+ * {n} $\leq$ 10
+ * 1 $\leq {A_{i}} \leq$ 1000,000,000
+ *
+ * ### Algorithm
+ * For each Mask 0 to 111....(2$\cdot${N} times) Add new pair to this set from
+ * yet unselected elements and try to maximize the newly formed mask after
+ * adding these elements with value formed after adding new pair.
+ * the ans we are looking for will be in mask which shows all elements are
+ * added, that is 111...(2$\cdot${N} times).
+ *
+ * @author [Syed Faizan](https://github.com/faizan2700)
+ */
+#include <algorithm>  /// for calculating gcd
+#include <array>      /// for std::array
+#include <cassert>    /// for using assert in self-test
+#include <iostream>   /// for IO operations
+#include <vector>     /// for std::vector
+
+/**
+ * @namespace dynamic_programming
+ * @brief Dynamic Programming algorithms
+ */
+namespace dynamic_programming {
+/**
+ * @namespace dp_bitmask
+ * @brief Functions for [Bitmasking DP
+ * approach](https://codeforces.com/blog/entry/337) implementation
+ */
+namespace dp_bitmask {
+/**
+ * @brief Find the maximum value of a new mask by adding
+ * a pair to a previously formed mask
+ * @tparam T size of the array `A`
+ * @param A array from which pairs will be made
+ * @param N array size
+ * @return maximum score which can be formed with pairs of this array
+ */
+template <size_t T>
+uint64_t find_max_score(const std::array<uint32_t, T>& A, const uint8_t N) {

```

Comment:
I think this should be passed by reference.
```suggestion
uint64_t find_max_score(const std::array<uint32_t, T>& A, const uint8_t& N) {
```